docs: connector orchestration (#3148)
feldera/feldera: docs/connectors/index.mdx

See [Delta Lake output connector documentation](/connectors/sinks/delta)
for an example of configuring the output buffer.

## Orchestration (starting and pausing)

:::note

Only input connectors currently support orchestration.

:::

> 

Reviewer: ```suggestion
Connector Orchestration enables users to activate or deactivate input connectors on demand, giving them control over the timing and order of data ingestion from multiple sources. It can, for example, be used to backfill a pipeline with historical data from a database or data lake before switching over to real-time ingestion from a streaming source like Kafka.
```

is the input connector active. The following table illustrates this:
```text
Pipeline state    Connector state    Connector is active?
--------------    ---------------    --------------------
Paused            Paused             No
Paused            Running            No
Running           Paused             No
Running           Running            Yes
```

### Orchestration example

1. Create and start a pipeline named `example` with the following SQL:
   ```sql
   CREATE TABLE numbers (
     num INT
   ) WITH (
       'connectors' = '[
           {
               "name": "c1",
               "paused": true,
               "transport": {
                   "name": "datagen",
                   "config": {"plan": [{ "rate": 1, "fields": { "num": { "range": [0, 10], "strategy": "uniform" } } }]}
               }
           },
           {
               "name": "c2",
               "paused": false,
               "transport": {
                   "name": "datagen",
                   "config": {"plan": [{ "rate": 1, "fields": { "num": { "range": [10, 20], "strategy": "uniform" } } }]}
               }
           }
       ]'
   );
   ```

> 

Reviewer: ```suggestion
Note that the `numbers` table has two input connectors, one of which has `paused` property set to `true`. This connector will be created in the `Paused` state.
```

> 2. Observe that although the pipeline is `Running`, the Changes Stream tab in the Web Console
>    (be sure to check the `numbers` table checkbox) only shows input records from connector `c2`
>    (i.e., `[10, 20)`) but not of connector `c1` (i.e., `[0, 10)`).

Reviewer: ```suggestion
2. Check the `numbers` table checkbox in the Change Stream tab. Observe that although the pipeline is `Running`, the change stream only shows input records from connector `c2` (i.e., `[10, 20)`) but not of connector `c1` (i.e., `[0, 10)`).
```
